### DISCUSSION TITLE
version 16 do not support arm6l platforms?

### DISCUSSION THREAD

**Discussion by bertlea** [nodejs/help#4854] [Q&A] [answered]
Hello, can any one confirm the arm6 is not longer supported after version 15?  I cannot find it in version 16's unofficial build downloads.   If there is any announcement or discussion confirmed that decision, it will be nice to share the link.   My Raspberry Pi Zeros are all running Node.js version 14's arm6 version.  It will be nice to confirm if I can upgrade them to version 16.

**Comment by richardlau** [comment, marked as answer]
arm6l has been unsupported since Node.js 12 (https://github.com/nodejs/build/issues/1677#issuecomment-486204349), with Node.js 10 End-of-Life this project no longer builds arm6l binaries as part of the regular releases.

The unoffiical builds (https://unofficial-builds.nodejs.org/) are just that, unofficial, and are not guaranteed to continue working without volunteers to keep it going. There's no decisions/announcements/considerations for these outside of https://github.com/nodejs/unofficial-builds.

Node.js 16 made changes to the required build chain, including dropping support for Python 2 which broke all of the unofficial builds (https://github.com/nodejs/unofficial-builds/issues/28). From the build logs in https://unofficial-builds.nodejs.org/logs/202104220522-v16.0.0/ it looks like the Docker file for arm6l has Python 3 but not a new enough version to be allowed by the configure scripts.

**Reply by bertlea** [reply]
@richardlau ,  Thank you for the quick answer!    So it is due to some technical reason prevented v16 unofficial builds to include arm6l currently.  I got it.

Is there an official or a good document can guide user to attempt build arm6l version of Node.js so that I can give it a try?

**Reply by richardlau** [reply]
Not that I know of. You can find the Docker file and shell script used https://github.com/nodejs/unofficial-builds/tree/master/recipes/armv6l -- it uses the cross compiler from https://github.com/rvagg/rpi-newer-crosstools.

**Reply by richardlau** [reply]
Got the armv6l builds working for Node.js 16. 
Node.js 16.10.0: https://unofficial-builds.nodejs.org/download/release/v16.1.0/

- https://unofficial-builds.nodejs.org/download/release/v16.1.0/node-v16.1.0-linux-armv6l.tar.gz
- https://unofficial-builds.nodejs.org/download/release/v16.1.0/node-v16.1.0-linux-armv6l.tar.xz

**Reply by bertlea** [reply]
Thank you very much @richardlau !   Just tried on one of my Pi-zero and it looks great.  Will upgrade my other Pi-Zeros to v16.
